src/toffy/image_stitching.py

```diff
@@ -218,7 +265,81 @@ def stitch_images(
                 channels=[chan],
                 max_image_size=max_img_size,
             )
+            if img_size_scale:
+                image_data = rescale_stitched_array(image_data, img_size_scale)
+
             fname = os.path.join(stitched_subdir, chan + "_stitched.tiff")
             stitched = data_utils.stitch_images(image_data, num_cols)
-            current_img = stitched.loc["stitched_image", :, :, chan].values / scale
+            current_img = stitched.loc["stitched_image", :, :, chan].values / intensity_scale
```

Comment:
No need for `.values` on the `xarray`.  `numpy` functions can be computed on the `xarray` object itself due to [`duck arrays`](https://docs.xarray.dev/en/latest/user-guide/duckarrays.html).